task22/js/main.js

```diff
@@ -1,151 +1,200 @@
+    for (const column of Object.keys(userTableColumn)) {
+      const tableDataElement = createElementWithClassName("td", "table-data");
+      tableDataElement.textContent = user[column];
+      tableBodyFragment
+        .appendChild(tableRowElement)
+        .appendChild(tableDataElement);
     }
-    tbodyElement.appendChild(tableBodyFragment);
-    return tbodyElement;
+  }
+  tbodyElement.appendChild(tableBodyFragment);
+  return tbodyElement;
 };
 
-const getCreatedSortButton = () => {
-    const sortButton = document.createElement("button");
-    const sortImage = createElementWithClassName("img", "sort-image");
-    sortButton.id = "js-sortButton";
-    sortImage.dataset.status = "default";
-    sortButton.appendChild(sortImage);
-    return sortButton;
+const getCreatedSortButtons = (columnKey) => {
+  const sortButton = createElementWithClassName("button", "js-sortButton");
+  const sortImage = createElementWithClassName("img", "sort-image");
+  sortButton.id = `${columnKey}`;
+  sortImage.dataset.status = "default";
+  sortImage.src = "../img/both.svg";
+  sortImage.alt = "both-image";
+  sortButton.appendChild(sortImage);
+  return sortButton;
 };
 
-const addEventForSortButton = (userContentsData) => {
-    const sortButton = document.getElementById("js-sortButton");
-    sortButton.addEventListener('click', (e) => {
-        const nextStatus = changeSortStatus(e.target.dataset.status);
-        e.target.dataset.status = nextStatus;
+const changeSortStatus = (status) => {
+  switch (status) {
+    case "default":
+      return "asc";
+    case "asc":
+      return "desc";
+    case "desc":
+    default:
+      return "default";
+  }
+};
 
-        sortContents(userContentsData, nextStatus);        
-    })
-}
+const defaultSettingForSortButton = (target, nextStatus) => {
+  const notTarget = [...document.querySelectorAll(".sort-image")].filter(
+    (column) => column !== target
+  );
+  notTarget.forEach((column) => {
+    if (nextStatus !== "default") {
+      column.dataset.status = "default";
+      column.src = "../img/both.svg";
+      column.alt = "both-image";
+    }
+  });
+};
 
-const changeSortStatus = (status) => {
-    switch(status){
-        case "default":
-            return "asc";
-        case "asc":
-            return "desc";
-        case "desc":
-        default:
-            return "default";
+const configForSortButton = (target, nextStatus) => {
+  const targetButton = [...document.querySelectorAll(".sort-image")].filter((column) => column === target);
+  targetButton.forEach((column) => {
+    if(nextStatus === "asc") {
+      column.src = "../img/asc.svg";
+      column.alt = "asc-image";
+      return;
     }
+    if(nextStatus === "desc") {
+      column.src = "../img/desc.svg";
+      column.alt = "desc-image";
+      return;
+    }
+    if(nextStatus === "default"){
+      column.src = "../img/both.svg";
+      column.alt = "both-image";
+      return;
+    }
+  })
```

Comment:
Q.
targetButtonはすでに引数で渡ってきているので、これでも問題ないでしょうか？
また、ifの場合だと最後のreturnは不要かなと思いました。

```suggestion
const configForSortButton = (target, nextStatus) => {
    if (nextStatus === "asc") {
      target.src = "../img/asc.svg";
      target.alt = "asc-image";
    return;
    }
    if (nextStatus === "desc") {
      target.src = "../img/desc.svg";
      target.alt = "desc-image";
      return;
    }
    if (nextStatus === "default") {
      target.src = "../img/both.svg";
      target.alt = "both-image";
    }
})
```

Reply:
おっしゃる通りです！！ご指摘ありがとうございます。難しく考え過ぎていました。
switch文に書き換えました。何度も確認いただいてありがとうございます。よろしくお願いします。
[8836028](https://github.com/seeeen044/JavaScript_handson/pull/24/commits/883602815cd734930f3d7861874bef83873da871)